wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -613,16 +606,14 @@ object Wallet extends WalletLogger {
     // safe since we're deriving from a priv
     val xpub = keyManager.deriveXPub(account).get
     val accountDb = AccountDb(xpub, account)
-    val accountDAO = wallet.accountDAO
 
     //see if we already have this account in our database
     //Three possible cases:
     //1. We have nothing in our database, so we need to insert it
     //2. We already have this account in our database, so we do nothing
     //3. We have this account in our database, with a DIFFERENT xpub. This is bad. Fail with an exception
     //   this most likely means that we have a different key manager than we expected
-    val accountOptF = accountDAO.read(account.coin, account.index)
-    accountOptF.flatMap {
+    wallet.accountDAO.read(account.coin, account.index).flatMap {
```

Comment:
again, I don't think the solution is to NOT use intermediate `val`

Reply:
It is